buildspec.yml

```diff
@@ -14,9 +14,28 @@ phases:
       # List the docker images 
       - docker images
 
-      # Push the image to ECR in the same account and same region the pipeline is hosted.
-      - docker tag amazon/aws-for-fluent-bit:latest amazon/aws-for-fluent-bit-test:latest
-      - ecs-cli push amazon/aws-for-fluent-bit-test:latest
+      # Push the image to ECR with corresponding architecture as the tag.
+      - aws ecr get-login-password --region ${AWS_REGION}| docker login --username AWS --password-stdin ${AWS_ACCOUNT}.dkr.ecr.${AWS_REGION}.amazonaws.com
+      - aws ecr create-repository --repository-name amazon/aws-for-fluent-bit-test --image-scanning-configuration scanOnPush=true --region ${AWS_REGION}  || true
+      - architecture=$(docker inspect --format='{{.Architecture}}'  amazon/aws-for-fluent-bit)
```

Comment:
I am not super clear here. For the first time, we only have AMD version image in dockerhub. How will we get the proper ARM architecture using this command. Excuse my ignorance with docker inspect.

Reply:
So in the build stage, we have two instances AMD and ARM running in parallel. Both instances will build the image separately, in parallel. Before these images are pushed to ECR, I tag each image with it's underlying architecture using  `architecture=$(docker inspect --format='{{.Architecture}}'  amazon/aws-for-fluent-bit)`. So as you can see, order does not matter here. 

However, when we create the manifest list after this step, we need to wait for both the images to be available in ECR which is why after each command, I have a `|| True`. This ensures build does not fail.